engine/src/main/java/org/terasology/engine/subsystem/lwjgl/LwjglGraphics.java

```diff
@@ -254,22 +255,85 @@ public Subtexture buildAsset(AssetUri uri, SubtextureData data) {
     }
 
     private void checkOpenGL() {
-        boolean canRunGame = GLContext.getCapabilities().OpenGL11
-                & GLContext.getCapabilities().OpenGL12
-                & GLContext.getCapabilities().OpenGL14
-                & GLContext.getCapabilities().OpenGL15
-                & GLContext.getCapabilities().GL_ARB_framebuffer_object
-                & GLContext.getCapabilities().GL_ARB_texture_float
-                & GLContext.getCapabilities().GL_ARB_half_float_pixel
-                & GLContext.getCapabilities().GL_ARB_shader_objects;
-
-        if (!canRunGame) {
-            String message = "Your GPU driver is not supporting the mandatory versions or extensions of OpenGL. Considered updating your GPU drivers? Exiting...";
-            logger.error(message);
-            JOptionPane.showMessageDialog(null, message, "Mandatory OpenGL version(s) or extension(s) not supported", JOptionPane.ERROR_MESSAGE);
-            throw new RuntimeException(message);
+        boolean[] requiredCapabilities = {
+                GLContext.getCapabilities().OpenGL11,
+                GLContext.getCapabilities().OpenGL12,
+                GLContext.getCapabilities().OpenGL14,
+                GLContext.getCapabilities().OpenGL15,
+                GLContext.getCapabilities().GL_ARB_framebuffer_object,
+                GLContext.getCapabilities().GL_ARB_texture_float,
+                GLContext.getCapabilities().GL_ARB_half_float_pixel,
+                GLContext.getCapabilities().GL_ARB_shader_objects};
+
+        requiredCapabilities[3] = false;
+
+        String[] capabilityNames = {"OpenGL11",
+                                    "OpenGL12",
+                                    "OpenGL14",
+                                    "OpenGL15",
+                                    "GL_ARB_framebuffer_object",
+                                    "GL_ARB_texture_float",
+                                    "GL_ARB_half_float_pixel",
+                                    "GL_ARB_shader_objects"};
+
+        boolean canRunTheGame = true;
+        String missingCapabilitiesMessage = "";
+
+        for (int index = 0; index < requiredCapabilities.length; index++) {
+            if (!requiredCapabilities[index]) {
+                missingCapabilitiesMessage += "    - " + capabilityNames[index] + "\n";
+                canRunTheGame = false;
+            }
         }
 
+        if (!canRunTheGame) {
+            String completeErrorMessage = completeErrorMessage(missingCapabilitiesMessage);
+            logger.error(completeErrorMessage);
+            JOptionPane.showMessageDialog(null, completeErrorMessage, "Required OpenGL version(s) or extension(s) not supported", JOptionPane.ERROR_MESSAGE);
```

Comment:
You need to call AWT methods from the EventDispatchThread using SwingUtilities.invoke.. Otherwise the splashscreen which might run in the background crashes with an IllegalStateEx.

I'd remove this line entirely as it is not the job of `LwjglGraphics` to show nice GUI dialogs. If you

     throw new IllegalStateException(completeErrorMessage);

instead, it will bubble up to `main()` and will be reported by CrashManager (if not headless, CR enabled, etc).